### PR TITLE
CHANGELOG: release 0.30.0

### PR DESCRIPTION
**Pull request by smallhive**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Not accurate error message for invalid authenticated requests (#882)
+- Return correct error on lock configuration check (#888)
+- Use older dev-env for GH tests (#919)
+- SDK to use slicing optimization (#923)
+- Update workflows (#930)
+- Don't use neofs-crypto directly in neofs/tree (#932)
+
+### Removed
+- 'v' from app version (#922)
+- Remove re-slicing for multipart (#931)
```
Comment (roman-khimov) [inline CHANGELOG.md:50]: `Removed Remove`

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
 
 ## [Unreleased]
 
+## [0.20.0] - 2024-03-26
```
Comment (roman-khimov) [inline CHANGELOG.md:7]: 0.30.0

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+
 ### Added
-- Passing an empty meta parameter value raises "Your metadata headers are not supported." error.
+- Passing an empty meta parameter value raises "Your metadata headers are not supported." error (#848)
+- Upload s3-tests and docker logs to neofs. (#836)
```
Comment (roman-khimov) [inline CHANGELOG.md:11]: This doesn't affect deliverables, so there is no need to mention it.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Minimal doc about local labels (#918)
+- Usage of gh-push-to-neofs action to test workflows (#920)
+- Return unimplemented error in PutBucketLogging (#925)
+- Return error if no tags in bucket (#924)
+- Return correct error on invalid grantee type (#926)
+- Sort parts by number and server creation time (#929)
 
 ### Changed
 - Go 1.20+ is required to build now, using 1.22 by default (#928)
+- GH workflow (#838, #837)
```
Comment (roman-khimov) [inline CHANGELOG.md:32]: Also.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
 
 ## [Unreleased]
 
+## [0.20.0] - 2024-03-26
+
 ### Added
-- Passing an empty meta parameter value raises "Your metadata headers are not supported." error.
+- Passing an empty meta parameter value raises "Your metadata headers are not supported." error (#848)
```
Comment (roman-khimov) [inline CHANGELOG.md:10]: Maybe better for the `Changed` section?

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Handle bad user id more gracefully (#879)
+
+### Updated
+- Documentation about ETag (#856)
+- Use SDK with pool base multi sessions (#855)
+- Public read acl results in full control permission (#866)
+- Documentation about Authenticated Users group (#871)
+- Not accurate error message for invalid authenticated requests (#882)
+- Return correct error on lock configuration check (#888)
+- Use older dev-env for GH tests (#919)
```
Comment (roman-khimov) [inline CHANGELOG.md:43]: Doesn't affect deliverables.

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
 ## [Unreleased]
 
+## [0.20.0] - 2024-03-26
+
 ### Added
-- Passing an empty meta parameter value raises "Your metadata headers are not supported." error.
+- Passing an empty meta parameter value raises "Your metadata headers are not supported." error (#848)
+- Upload s3-tests and docker logs to neofs. (#836)
+- Preallocate buffers to object uploads (#835)
+- Sdk optimizations (#839)
```
Comment (roman-khimov) [inline CHANGELOG.md:13]: `Changed`

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
 ### Changed
 - Go 1.20+ is required to build now, using 1.22 by default (#928)
+- GH workflow (#838, #837)
+- Limit objects amount for DeleteObjects (#849)
+- Handle bad user id more gracefully (#879)
+
+### Updated
+- Documentation about ETag (#856)
+- Use SDK with pool base multi sessions (#855)
+- Public read acl results in full control permission (#866)
```
Comment (roman-khimov) [inline CHANGELOG.md:39]: `Fixed` section

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Return unimplemented error in PutBucketLogging (#925)
+- Return error if no tags in bucket (#924)
+- Return correct error on invalid grantee type (#926)
+- Sort parts by number and server creation time (#929)
 
 ### Changed
 - Go 1.20+ is required to build now, using 1.22 by default (#928)
+- GH workflow (#838, #837)
+- Limit objects amount for DeleteObjects (#849)
+- Handle bad user id more gracefully (#879)
```
Comment (roman-khimov) [inline CHANGELOG.md:34]: `Fixed`

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
 
+## [0.20.0] - 2024-03-26
+
 ### Added
-- Passing an empty meta parameter value raises "Your metadata headers are not supported." error.
+- Passing an empty meta parameter value raises "Your metadata headers are not supported." error (#848)
+- Upload s3-tests and docker logs to neofs. (#836)
+- Preallocate buffers to object uploads (#835)
+- Sdk optimizations (#839)
+- Restrict presigned URL lifetime (#883)
```
Comment (roman-khimov) [inline CHANGELOG.md:14]: `Fixed`. We're not adding  a lot, btw.

File: CHANGELOG.md [hunk 11]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Return unsupported error on GetObjectTorrent (#884)
+- Return unsupported error on GetBucketPolicyStatus (#886)
+- Return unsupported error on PutPublicAccessBlock (#887)
+- Return the correct error, if x-amz-credential is not provided (#892)
```
Comment (roman-khimov) [inline CHANGELOG.md:18]: Fixed 4 above.

File: CHANGELOG.md [hunk 12]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- GH workflow (#838, #837)
+- Limit objects amount for DeleteObjects (#849)
+- Handle bad user id more gracefully (#879)
+
+### Updated
+- Documentation about ETag (#856)
+- Use SDK with pool base multi sessions (#855)
+- Public read acl results in full control permission (#866)
+- Documentation about Authenticated Users group (#871)
+- Not accurate error message for invalid authenticated requests (#882)
```
Comment (roman-khimov) [inline CHANGELOG.md:41]: `Fixed`

File: CHANGELOG.md [hunk 13]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Passing an empty meta parameter value raises "Your metadata headers are not supported." error (#848)
+- Upload s3-tests and docker logs to neofs. (#836)
+- Preallocate buffers to object uploads (#835)
+- Sdk optimizations (#839)
+- Restrict presigned URL lifetime (#883)
+- Return unsupported error on GetObjectTorrent (#884)
+- Return unsupported error on GetBucketPolicyStatus (#886)
+- Return unsupported error on PutPublicAccessBlock (#887)
+- Return the correct error, if x-amz-credential is not provided (#892)
+- Check x-amz-expires header on maximum border (#893)
```
Comment (roman-khimov) [inline CHANGELOG.md:19]: `Fixed`

File: CHANGELOG.md [hunk 14]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Return unimplemented error on GetPublicAccessBlock (#896)
```
Comment (roman-khimov) [inline CHANGELOG.md:20]: Fixed

File: CHANGELOG.md [hunk 15]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Limit objects amount for DeleteObjects (#849)
+- Handle bad user id more gracefully (#879)
+
+### Updated
+- Documentation about ETag (#856)
+- Use SDK with pool base multi sessions (#855)
+- Public read acl results in full control permission (#866)
+- Documentation about Authenticated Users group (#871)
+- Not accurate error message for invalid authenticated requests (#882)
+- Return correct error on lock configuration check (#888)
```
Comment (roman-khimov) [inline CHANGELOG.md:42]: `Fixed`

File: CHANGELOG.md [hunk 16]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Sdk optimizations (#839)
+- Restrict presigned URL lifetime (#883)
+- Return unsupported error on GetObjectTorrent (#884)
+- Return unsupported error on GetBucketPolicyStatus (#886)
+- Return unsupported error on PutPublicAccessBlock (#887)
+- Return the correct error, if x-amz-credential is not provided (#892)
+- Check x-amz-expires header on maximum border (#893)
+- Return unimplemented error on GetPublicAccessBlock (#896)
+- Support for aws-chunked (#914)
+- Store bucket owner pub key in container attributes (#915)
```
Comment (roman-khimov) [inline CHANGELOG.md:22]: `Changed`, but what's more interesting is how it affects behavior.

File: CHANGELOG.md [hunk 17]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Minimal doc about local labels (#918)
```
Comment (roman-khimov) [inline CHANGELOG.md:23]: Internal thing, doesn't affect deliverable.

File: CHANGELOG.md [hunk 18]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Usage of gh-push-to-neofs action to test workflows (#920)
```
Comment (roman-khimov) [inline CHANGELOG.md:24]: Internal.

File: CHANGELOG.md [hunk 19]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Return unimplemented error in PutBucketLogging (#925)
+- Return error if no tags in bucket (#924)
+- Return correct error on invalid grantee type (#926)
```
Comment (roman-khimov) [inline CHANGELOG.md:27]: Fixed three above.

File: CHANGELOG.md [hunk 20]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+- Sort parts by number and server creation time (#929)
```
Comment (roman-khimov) [inline CHANGELOG.md:28]: What's the real problem here?

File: CHANGELOG.md [hunk 21]
```diff
@@ -4,11 +4,50 @@ This document outlines major changes between releases.
...
+### Updated
+- Documentation about ETag (#856)
+- Use SDK with pool base multi sessions (#855)
+- Public read acl results in full control permission (#866)
+- Documentation about Authenticated Users group (#871)
+- Not accurate error message for invalid authenticated requests (#882)
+- Return correct error on lock configuration check (#888)
+- Use older dev-env for GH tests (#919)
+- SDK to use slicing optimization (#923)
+- Update workflows (#930)
```
Comment (roman-khimov) [inline CHANGELOG.md:45]: Internal

**Review by roman-khimov — CHANGES_REQUESTED** [review]
I don't see #851, #867, #880, #889, #894, #890, #876, #917, #935 mentioned and #933 has some updates as well. Dependency changes are significant, they affect our product.

Please update VERSION file as well.

But most importantly, we can't make a release if our build is broken.

File: CHANGELOG.md [hunk 22]
```diff
@@ -4,11 +4,57 @@ This document outlines major changes between releases.
 
 ## [Unreleased]
 
+## [0.30.0] - 2024-03-26
```
Comment (roman-khimov) [inline CHANGELOG.md:7]: 27th

File: CHANGELOG.md [hunk 23]
```diff
@@ -4,11 +4,57 @@ This document outlines major changes between releases.
...
+- build(deps): bump golang.org/x/net from 0.14.0 to 0.17.0 (#851)
+- build(deps): bump github.com/nats-io/nats-server/v2 from 2.7.4 to 2.9.23 (#867)
+- build(deps): bump google.golang.org/grpc from 1.57.0 to 1.57.1 (#880)
+- build(deps): bump github.com/nats-io/nkeys from 0.4.4 to 0.4.6 (#894)
+- build(deps): bump golang.org/x/crypto from 0.14.0 to 0.17.0 (#917)
+- build(deps): bump google.golang.org/protobuf from 1.31.0 to 1.33.0 (#935)
+
+### Removed
+- Removed 'v' from app version (#922)
+- Removed re-slicing for multipart (#931)
```
Comment (roman-khimov) [inline CHANGELOG.md:43]: "Removed removed" twice now?

File: CHANGELOG.md (hunk 15 above)
Comment (roman-khimov) [reply CHANGELOG.md:31]: Still in `updated`

File: CHANGELOG.md (hunk 14 above)
Comment (roman-khimov) [reply CHANGELOG.md:12]: Not `added`, but `fixed`

File: CHANGELOG.md [hunk 24]
```diff
@@ -4,11 +4,57 @@ This document outlines major changes between releases.
...
+## [0.30.0] - 2024-03-26
+
 ### Added
-- Passing an empty meta parameter value raises "Your metadata headers are not supported." error.
+- Preallocate buffers to object uploads (#835)
+- Check x-amz-expires header on maximum border (#893)
+- Return unimplemented error on GetPublicAccessBlock (#896)
+- Support for aws-chunked (#914)
+- Return unimplemented error in PutBucketLogging (#925)
+- Return not supported error for PutObjectLegalHold command if OFF state (#889)
```
Comment (roman-khimov) [inline CHANGELOG.md:15]: These two are also `fixed`

File: CHANGELOG.md (hunk 13 above)
Comment (roman-khimov) [reply CHANGELOG.md:11]: Fixed.

File: CHANGELOG.md (hunk 23 above)
Comment (roman-khimov) [reply CHANGELOG.md:37]: There is no point in writing "removed" in "removed" section. Consider
```
Removed:
 * this
 * that
```
and
```
Removed:
 * removed this
 * removed that
```
Same goes for "fixed fixed"/"added added"/etc.

Comment (smallhive) [reply CHANGELOG.md:37]: Ow, now I see. Just misunderstood, the first comment was `Removed Remove` for two rows. I decided it was suggestion for row changing

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**